backends/advanced/src/advanced_omi_backend/routers/modules/system_routes.py

```
@@ -128,6 +129,53 @@ async def delete_all_user_memories(current_user: User = Depends(current_active_u
+async def save_chat_config(
+    request: Request,
+    current_user: User = Depends(current_superuser)
+):
+    """Save chat configuration from YAML. Admin only."""
+    try:
+        yaml_content = await request.body()
+        yaml_str = yaml_content.decode('utf-8')
+        result = await system_controller.save_chat_config_yaml(yaml_str)
+        return JSONResponse(content=result)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)